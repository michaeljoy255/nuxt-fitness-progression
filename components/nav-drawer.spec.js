import { mount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'
import EmptyStub from '../test/stubs/empty-stub'
import NavDrawer from './nav-drawer'

const localVue = createLocalVue()
let wrapper

beforeEach(() => {
  const vuetify = new Vuetify()
  wrapper = mount(NavDrawer, {
    stubs: {
      NavDrawerListItem: EmptyStub,
    },
    mocks: {
      $store: {
        state: {
          ui: {
            navDrawer: false,
          },
        },
      },
    },
    localVue,
    vuetify,
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('Component > nav-drawer', () => {
  test('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
