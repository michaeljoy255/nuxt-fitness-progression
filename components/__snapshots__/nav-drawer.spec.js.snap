// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component > nav-drawer matches the snapshot 1`] = `
"<nav class=\\"v-navigation-drawer v-navigation-drawer--clipped v-navigation-drawer--close v-navigation-drawer--fixed v-navigation-drawer--is-mobile theme--light\\" style=\\"height: 100vh; top: 0px; transform: translateX(-100%); width: 256px;\\">
  <div class=\\"v-navigation-drawer__content\\">
    <div class=\\"v-list v-sheet theme--light v-list--nav\\">
      <div route=\\"/\\" icon=\\"dashboard\\" name=\\"Dashboard\\"></div>
      <hr role=\\"separator\\" aria-orientation=\\"horizontal\\" class=\\"mb-2 v-divider theme--light\\">
      <div route=\\"/test\\" icon=\\"timeline\\" name=\\"Charts\\"></div>
      <div route=\\"/test\\" icon=\\"assignment\\" name=\\"Workouts\\"></div>
      <div route=\\"/test\\" icon=\\"fitness_center\\" name=\\"Exercises\\"></div>
      <div route=\\"/test\\" icon=\\"straighten\\" name=\\"Measurements\\"></div>
      <div route=\\"/test\\" icon=\\"sd_card\\" name=\\"Records\\"></div>
      <hr role=\\"separator\\" aria-orientation=\\"horizontal\\" class=\\"mb-2 v-divider theme--light\\">
      <div route=\\"/test\\" icon=\\"info\\" name=\\"About\\"></div>
      <div route=\\"/test\\" icon=\\"announcement\\" name=\\"Test\\"></div>
    </div>
  </div>
  <div class=\\"v-navigation-drawer__border\\"></div>
</nav>"
`;
